docs/source/tutorials/introductory/logging-first-model/step6-logging-a-run.rst

```diff
@@ -0,0 +1,204 @@
+        from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score
+
+Notice that here we aren't importing the ``MlflowClient`` directly. For this portion, we're going to
+be using the ``fluent`` API. The fluent APIs use a globally referenced state of the MLflow tracking
+server's uri. This global instance allows for us to use these 'higher-level' (simpler) APIs to perform
+every action that we can otherwise do with the ``MlflowClient``, with the addition of some other useful
+syntax (such as context handlers that we'll be using very shortly) to make integrating MLflow to
+ML workloads as simple as possible.
+
+In order to use the ``fluent`` API, we'll need to set the global reference to the Tracking server's
+address. We do this via the following command:
+
+.. code-section::
+    .. code-block:: python
+        :caption: Set the tracking uri
+
+        mlflow.set_tracking_uri("http://127.0.0.1:8080")
+
+Once this is set, we can define a few more constants that we're going to be using when logging our
+training events to MLflow in the form of runs. We'll start by defining an Experiment that will be used
+to log runs to. The parent-child relationship of Experiments to Runs and its utility will become very
+clear once we start iterating over some ideas and need to compare the results of our tests.
+
+.. code-section::
+    .. code-block:: python
+        :caption: Set the tracking uri
+
+        # Sets the current active experiment to the "Apple_Models" experiment and
+        # returns the Experiment metadata
+        apple_experiment = mlflow.set_experiment("Apple_Models")
+
+        # Define a run name for this iteration of training.
+        # If this is not set, a unique name will be auto-generated for your run.
+        run_name = "apples_rf_test"
+
+        # Define an artifact path that the model will be saved to.
+        artifact_path = "rf_apples"
+
+With these variables defined, we can commence with actually training a model.
+
+Firstly, let's look at what we're going to be running. Following the code display, we'll look at
+an annotated version of the code.
+
+.. code-section::
+    .. code-block:: python
+        :caption: Use MLflow to log the training of a model
+
+        with mlflow.start_run(run_name=run_name) as run:
+            # Split the data into features and target and drop irrelevant date
+            # field and target field
+            X = data.drop(columns=["date", "demand"])
+            y = data["demand"]
+
+            # Split the data into training and validation sets
+            X_train, X_val, y_train, y_val = train_test_split(X, y, test_size=0.2, random_state=42)
+
+            params = {
+                "n_estimators": 100,
+                "max_depth": 6,
+                "min_samples_split": 10,
+                "min_samples_leaf": 4,
+                "bootstrap": True,
+                "oob_score": False,
+                "random_state": 888
+            }
```

Comment:
I'd reorganize the code like this:

```python
params = ...
metrics = ...
model = ...

with mlflow.start_run(run_name=run_name) as run:
    mlflow.log_params(params)
    mlflow.log_metrics(metrics)
    mlflow.sklearn.log_model(
        sk_model=rf,
        input_example=X_val,
        artifact_path=artifact_path
    )
```

Reply:
good idea. Changed in all places (docs and notebooks)